Vizzuality/marxan-cloud
# Feature shapefile upload

## Feature shapefile upload

### Overview

This PR adds a new endpoint `POST /api/v1/projects/:id/features/shapefile` that allows users to upload feature shapefiles.

To use this endpoint, you'll need to provide a `.zip` file containing the different format files that compose a shapefile, and send it as `multipart/form-data` to the endpoint specified above.

The upload happens synchronously, so once you receive the response, the data should already be available in the database. Also, keep in mind there's a limit on the maximum size of the uploaded file, currently set to 100MB (this limit is configurable via env vars).

### Designs

[Figma](https://www.figma.com/file/hq0BZNB9fzyFSbEUgQIHdK/Marxan-Visual_V02?node-id=2991%3A2492)

_This PR does not contain the implementation of the UI for this feature, just the BE side of things._

### Testing instructions

Use the new endpoint and check that data gets correctly imported to the databases (API and geo) - check the JIRA ticket below for more details on where the data should be placed.

### Feature relevant tickets

[MARXAN-543](https://vizzuality.atlassian.net/browse/MARXAN-543)

---

## Checklist before submitting

      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
